src/client/scripts/game/chess/selection.js

```diff
@@ -199,9 +199,12 @@ const selection = (function() {
         legalMoves = legalmoves.calculate(game.getGamefile(), pieceSelected)
 
         const pieceColor = math.getPieceColorFromType(pieceSelected.type);
-        isOpponentPiece = onlinegame.areInOnlineGame() ? pieceColor !== onlinegame.getOurColor()
-                               /* Local Game */        : pieceColor !== game.getGamefile().whosTurn;
-        isPremove = !isOpponentPiece && onlinegame.areInOnlineGame() && !onlinegame.isItOurTurn();
+        if (onlinegame.areInOnlineGame()) isOpponentPiece = pieceColor !== onlinegame.getOurColor();
+        else if (enginegame.areInEngineGame()) isOpponentPiece = pieceColor !== enginegame.getOurColor();
+        else isOpponentPiece = pieceColor !== game.getGamefile().whosTurn;
```

Comment:
I'm seeing a lot of this repeat logic where to find out if it's our turn we have to check onlingegame, enginegame, and local game. What if we created a method in game.js called `isItOurTurn()` that does all this logic for us, no matter what kind of game we are in?